### PR TITLE
zig: upgrade to 0.6.0

### PR DESCRIPTION
**Pull request by Xe**
###### Motivation for this change

See [Zig 0.6.0 release notes](https://ziglang.org/download/0.6.0/release-notes.html)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/zig/default.nix [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
+    ./zig test $src/test/stage1/behavior.zig
+    runHook postCheck
+  '';
+
+  doCheck = true;
+
   meta = with stdenv.lib; {
-    description = "Programming languaged designed for robustness, optimality, and clarity";
+    description =
+      "Programming languaged designed for robustness, optimality, and clarity";
```
Comment (andrewrk) [inline pkgs/development/compilers/zig/default.nix:44]:
```suggestion
      "general-purpose programming language and toolchain for maintaining robust, optimal, and reusable software";
```

**Review by andrewrk — APPROVED** [review]

File: pkgs/development/compilers/zig/default.nix [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+    libxml2
+    zlib
+  ];
+
+  patches = [ ./llvm10_polly.patch ];
+
+  postPatch = ''
+    export llvm_extras=-Wl,${llvmPackages.llvm}/lib/LLVMPolly.so
+    substituteAllInPlace CMakeLists.txt
+  '';
```
Comment (cole-h) [inline pkgs/development/compilers/zig/default.nix:28]: I think something like this should work (will also need to add `substituteAll` to the args):

```nix
  patches = [
    (substituteAll {
        src = ./llvm10_polly.patch;
        llvm_extras = "-Wl,${llvmPackages.llvm}/lib/LLVMPolly.so";
    })
  ];
```

**Review by andrewrk — APPROVED** [review]

**Comment by andrewrk** [comment]
Why does it keep re-requesting my review?

**Review by cole-h — APPROVED** [review]
LGTM. `ldd $(which zig)` also shows `LLVMPolly.so`, so my suggestion doesn't appear to have screwed anything up ^^. Built and ran the "Hello World" example just fine.

```
[5 built, 10 copied (907.4 MiB), 119.7 MiB DL]
https://github.com/NixOS/nixpkgs/pull/85183
1 package built:
zig
```

**Comment by worldofpeace** [comment]
just for future reference, updates commit messages are always like
```
zig: 0.5.0 -> 0.6.0
```

See https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Comment by sikmir** [comment]
@andrewrk Build fails on darwin:
```sh
/private/var/folders/wf/qxrs82v946n91pb5s71n3dy80000gp/T/nix-build-zig-0.6.0.drv-0/source/src/os.cpp:1105:9: error: use of undeclared identifier 'futimens'
    if (futimens(file, times) == -1) {
        ^
1 error generated.
make[2]: *** [CMakeFiles/zigcompiler.dir/build.make:304: CMakeFiles/zigcompiler.dir/src/os.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:225: CMakeFiles/zigcompiler.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
builder for '/nix/store/b7w0k884rcg5f0nlwm1jahxazgiillbw-zig-0.6.0.drv' failed with exit code 2
error: build of '/nix/store/b7w0k884rcg5f0nlwm1jahxazgiillbw-zig-0.6.0.drv' failed
```
My macOS version:
```sh
$ sw_vers -productVersion
10.13.6
```
According to https://ziglang.org/#Support-Table, macOS 10.13+ is supported.

**Pull request merged**